agent/php_observer.c

```diff
@@ -75,13 +76,58 @@
 static zend_observer_fcall_handlers nr_php_fcall_register_handlers(
     zend_execute_data* execute_data) {
   zend_observer_fcall_handlers handlers = {NULL, NULL};
+  zend_string *func_name = NULL, *scope_name = NULL;
+  nruserfn_t* wr;
   if (NULL == execute_data) {
     return handlers;
   }
   if ((NULL == execute_data->func)
       || (ZEND_INTERNAL_FUNCTION == execute_data->func->type)) {
     return handlers;
   }
+
+  if (nrunlikely(NR_PHP_PROCESS_GLOBALS(special_flags).show_executes)) {
+    nr_php_show_exec("observe", execute_data, NULL);
+  }
+
+  if (OP_ARRAY_IS_A_FILE(NR_OP_ARRAY)) {
+    /*
+     * Let's get the framework info.
+     */
+    nr_php_execute_file(NR_OP_ARRAY, execute_data, NULL TSRMLS_CC);
+    return handlers;
+  }
+
+  if (OP_ARRAY_IS_A_METHOD(NR_OP_ARRAY)) {
+    scope_name = execute_data->func->op_array.scope->name;
+    func_name = execute_data->func->op_array.function_name;
+#if 0
+    zend_long cn_hash = zend_hash_func(ZSTR_VAL(scope_name), ZSTR_LEN(scope_name));
+    zend_long fn_hash = zend_hash_func(ZSTR_VAL(func_name), ZSTR_LEN(func_name));
+    nrl_always("class_name=%s, class_name->h=%zu, cn_hash=%zu, function_name=%s, function_name->h=%zu, fn_hash=%zu",
+                ZSTR_VAL(scope_name), scope_name->h, cn_hash,
+                ZSTR_VAL(func_name), func_name->h, fn_hash);
+#endif
+  } else if (OP_ARRAY_IS_A_FUNCTION(NR_OP_ARRAY)){
+    func_name = execute_data->func->op_array.function_name;
+#if 0
+    zend_long fn_hash = zend_hash_func(ZSTR_VAL(func_name), ZSTR_LEN(func_name));
+    nrl_always("function_name=%s, function_name->h=%zu, fn_hash=%zu", ZSTR_VAL(func_name), func_name->h, fn_hash);
```

Comment:
Probably want these log levels to be verbosedebug

Reply:
I plan on removing the code between `#if 0`.